pipelines/nextflow/workflows/dumper_pipeline/main.nf

```diff
@@ -38,16 +44,29 @@ if (params.brc_mode) {
     params.brc_mode = params.brc_mode as Integer
 }
 
+// Prepare the server params
+include { extract_url_args; generate_url } from '../../modules/utils/utils.nf'
 def create_server(params) {
-    server = [
-        "host": params.host,
-        "port": params.port,
-        "user": params.user,
-        "password": ""
-    ]
+    if (params.server_url && !params.host) {
+        server = extract_url_args(params.server_url)
+        server.url = params.server_url
+    }
+    if (params.host && !params.server_url) {
+        server = [
+            "host": params.host,
+            "port": params.port,
+            "user": params.user,
+        ]
     if (params.password) {
         server["password"] = params.password
     }
+        server.url = generate_url("mysql", params.host, params.port, params.user, params.password)
+    }
+    if (!server.url or !server.host) {
+        log.info paramsHelp("Server URL (--server_url) or parameters needed (--host, --port, --user, --password)")
```

Comment:
The previous code will do nothing if both URL and host are provided. I believe it may be easier/better to just provide a parameter precedence:
```suggestion
    if (params.server_url) {
        server = extract_url_args(params.server_url)
        server.url = params.server_url
    } else if (params.host) {
        server = [
            "host": params.host,
            "port": params.port,
            "user": params.user,
        ]
        if (params.password) {
            server["password"] = params.password
        }
        server.url = generate_url("mysql", params.host, params.port, params.user, params.password)
    } else {
        log.info paramsHelp(
            "Server URL (--server_url) or parameters needed (--host, --port, --user, --password)"
        )
```